src/codegen/generate.ts

```diff
@@ -756,12 +741,17 @@ export default class ApiGenerator {
     const propertyNames = Object.keys(props);
     const filteredPropertyNames = propertyNames.filter((name) => {
       const schema = props[name];
-      if (!onlyMode)
-        return (
-          !this.isSchemaReadOnly(schema) && !this.isSchemaWriteOnly(schema)
-        );
-      if (onlyMode === "readOnly") return this.isSchemaReadOnly(schema);
-      else if (onlyMode === "writeOnly") return this.isSchemaWriteOnly(schema);
+      const isReadOnly = this.checkSchemaOnlyMode(schema, "readOnly", false);
+      const isWriteOnly = this.checkSchemaOnlyMode(schema, "writeOnly", false);
+
+      switch (onlyMode) {
+        case "readOnly":
+          return isReadOnly || !isWriteOnly;
+        case "writeOnly":
+          return isWriteOnly || !isReadOnly;
+        default:
+          return !isReadOnly && !isWriteOnly;
+      }
```

Comment:
now essentially only filtering out read+write props for base types, write props for read types and read props for write types